aleph-client/src/rpc.rs

```diff
@@ -10,18 +11,95 @@ fn json_req(method: &str, params: Value, id: u32) -> Value {
     })
 }
 
-pub fn author_rotate_keys() -> Value {
+pub fn author_rotate_keys_json() -> Value {
     json_req("author_rotateKeys", Value::Null, 1)
 }
 
+fn state_query_storage_at_json(storage_keys: &Vec<StorageKey>) -> Value {
+    json_req(
+        "state_queryStorageAt",
+        Value::Array(vec![
+            Value::Array(
+                storage_keys
+                    .into_iter()
+                    .map(|storage_key| Value::String(hex::encode(storage_key)))
+                    .collect::<Vec<_>>(),
+            ),
+            Value::Null,
+        ]),
+        1,
+    )
+}
+
+fn parse_query_storage_at_result(
+    maybe_json_result: Option<String>,
+) -> Result<Vec<Option<String>>, String> {
+    match maybe_json_result {
+        None => Err(String::from(format!("Returned result was null!",))),
+        Some(result) => {
+            let mut storage_entries_in_hex_string = vec![];
+
+            let parsed_json_value: Value = serde_json::from_str(&result[..]).map_err(|_| {
+                String::from(&format!("Failed to parse result {:?} into JSON", result))
+            })?;
+            let storage_entries = match parsed_json_value[0]["changes"].as_array() {
+                Some(storage_entries) => storage_entries,
+                None => {
+                    return Err(String::from(&format!(
+                        "Parsed json value {:?} does not have \"changes\" element!",
+                        parsed_json_value
+                    )))
+                }
+            };
+            for entry in storage_entries {
+                let storage_key_and_entry = match entry.as_array() {
+                    Some(entry) => entry,
+                    None => {
+                        return Err(String::from(&format!(
+                            "Value {:?} is not an array!",
+                            entry.clone()
+                        )))
+                    }
+                };
+                if storage_key_and_entry.len() != 2 {
+                    return Err(String::from(&format!(
+                        "Expected {:?} to be length of 2!",
+                        storage_key_and_entry
+                    )));
+                }
+
+                let maybe_entry_in_hex_string = match storage_key_and_entry[1].as_str() {
```

Comment:
Nit: a pattern like that is nicely expressed by using `Option::map` (i.e. when you want to transform the value inside the `Some` and leave `None` as is.

Reply:
Great comments, thanks for that :) simplified code